src/selectedtests/task_mappings/create_task_mappings.py

```diff
@@ -23,6 +25,49 @@
 TASK_BUILDS_KEY = "builds"
 
 
+def generate_task_mappings(
+    evg_api: EvergreenApi,
```

Comment:
IMO would be simpler if this got refactored at some point to be a `TaskMapper` that is configured with things like `project`, `EvergreenApi`, `Repo` etc and knows how to create `TaskMappings`, and for `TaskMappings` to be a more simple data transfer class that doesn't need to be its own factory --> perhaps only knowing how to serialize itself for storage purposes.

Reply:
Perhaps this could work, I do think it would be a big refactor. If you feel strongly we could make a ticket for this.